<a href="https://colab.research.google.com/github/Phonchanit/python_dataviz2024/blob/main/Exam3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd

In [7]:
Data_Doctor = pd.read_csv('Data_Docter_ศ.นพ.ชวลิต.csv')
Data_Doctor.head()

name  \
0  ศ.นพ.ชวลิต ศูนย์มะเร็ง รพ.จุฬาลงกรณ์   
1                        Nubeer Kithiya   
2                       วาสนา คุ้มจิตต์   
3                  Wansook Chompoo Ploy   
4                           Aphit Aphit   

                                            comments  ยอดไลค์  จำนวนการตอบกลับ  
0  กดที่ลิงค์นี้เพื่อเข้าไปอ่านคอมเม้นของโพสต์ครั...      456                3  
1  เป็นมะเร็งลำไส้ใหญ่ ระยะที่ 3 ตอนอายุ 26 ค่ะ พ...     1400               70  
2  แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402               11  
3  #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...     1600               44  
4  ลูกสาวเป็นมะเร็งกระดูก Osteosarcomaช่วงเข่า อา...      480                5

In [8]:
Data_Doctor.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1653 entries, 0 to 1652
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   name             1653 non-null   object
 1   comments         1653 non-null   object
 2   ยอดไลค์          1653 non-null   int64 
 3   จำนวนการตอบกลับ  1653 non-null   int64 
dtypes: int64(2), object(2)
memory usage: 51.8+ KB


In [18]:
Data_Doctor.describe()

ยอดไลค์  จำนวนการตอบกลับ
count  1653.000000      1653.000000
mean      6.778584         0.586207
std      70.971719         3.097387
min       0.000000         0.000000
25%       0.000000         0.000000
50%       0.000000         0.000000
75%       0.000000         0.000000
max    1600.000000        70.000000

In [12]:
# กรองเฉพาะคอลัมน์ที่เป็นตัวเลข
numeric_data = Data_Doctor.select_dtypes(include=['float64', 'int64'])

# คำนวณค่าสหสัมพันธ์ระหว่างตัวแปรที่เป็นตัวเลข
correlation_matrix = numeric_data.corr()

# แสดงค่า correlation matrix
correlation_matrix

ยอดไลค์  จำนวนการตอบกลับ
ยอดไลค์          1.000000         0.748161
จำนวนการตอบกลับ  0.748161         1.000000

# **สรุป**

จากผลลัพธ์ที่ได้ในการคำนวณค่าสหสัมพันธ์ระหว่างตัวแปร ยอดไลค์ และ จำนวนการตอบกลับ เราสามารถสังเกตเห็นจุดที่น่าสนใจ ดังนี้

1. ความสัมพันธ์เชิงบวกระหว่างยอดไลค์และจำนวนการตอบกลับ
ค่าสหสัมพันธ์ที่ได้คือ 0.748161 ซึ่งเป็นค่าสหสัมพันธ์เชิงบวกที่ค่อนข้างสูง แสดงให้เห็นว่าเมื่อโพสต์แล้วจะได้รับยอดไลค์มากขึ้น ก็มีแนวโน้มที่จะมีจำนวนการตอบกลับเพิ่มขึ้นด้วย

2. อาจสะท้อนถึงพฤติกรรมของผู้ใช้
การที่ยอดไลค์และจำนวนการตอบกลับ มีความสัมพันธ์เชิงบวก อาจบ่งบอกถึงพฤติกรรมของผู้ใช้ในแพลตฟอร์มว่ามีการตอบกลับมากขึ้นเมื่อพวกเขาชอบเนื้อหาหรือโพสต์นั้นๆ
ข้อมูลนี้สามารถใช้ในการปรับปรุงการออกแบบโพสต์ให้ได้รับความสนใจจากผู้ใช้ไม่เพียงแค่กดไลค์ แต่ยังมีการตอบกลับหรือแสดงความคิดเห็นเพิ่มเติม


จุดที่น่าสนใจจากข้อมูลนี้แสดงให้เห็นถึงความสัมพันธ์ที่ชัดเจนระหว่างยอดไลค์และจำนวนการตอบกลับ ซึ่งสามารถนำไปใช้ในการวางแผนกลยุทธ์ทางการตลาดหรือการสื่อสารในอนาคต รวมถึงการวิเคราะห์พฤติกรรมผู้ใช้ในแพลตฟอร์ม